api/v1alpha1/mysqlaaduser_conversion.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package v1alpha1
+
+import (
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	"github.com/Azure/azure-service-operator/api/v1alpha2"
+)
+
+func (src *MySQLAADUser) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha2.MySQLAADUser)
+
+	// ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	// Spec
+	dst.Spec.ResourceGroup = src.Spec.ResourceGroup
+	dst.Spec.Server = src.Spec.Server
+	dst.Spec.AADID = src.Spec.AADID
+	dst.Spec.Username = src.Spec.Username
+
+	// All roles should be interpreted as database roles in v1alpha2
```

Comment:
minor: This says what you're doing, but so does the code. I think it makes more sense to expand on the why, for example: "because that was the only kind of role supported by v1alpha1"